lib/signaling/v2/peerconnection.js

```diff
@@ -865,10 +896,14 @@ class PeerConnectionV2 extends StateMachine {
         });
       }
     }).then(() => this._peerConnection.setRemoteDescription(description)).then(() => {
-      if (description.type === 'answer' && this._isRestartingIce) {
-        this._log.debug('An ICE restart was in-progress and is now completed');
-        this._isRestartingIce = false;
+      if (description.type === 'answer') {
+        if (this._isRestartingIce) {
+          this._log.debug('An ICE restart was in-progress and is now completed');
+          this._isRestartingIce = false;
+        }
+        negotiationCompleted(this);
       }
+
       if (description.type === 'answer' && this._isUnifiedPlan) {
```

Comment:
@makarandp0 If we enhance `negotiationCompleted` like I've suggested then, we can remove this `if` block.